### PR TITLE
don't connect module closers to destroy nodes

### LINKED ISSUE #26166: Cycle error when using create_before_destroy with depends_on referencing a module
**Version**
```
Terraform v0.13.0
+ provider registry.terraform.io/hashicorp/aws v3.4.0
+ provider registry.terraform.io/hashicorp/external v1.2.0
+ provider registry.terraform.io/hashicorp/null v2.1.2
```

**Usecase**
We have two modules A and B. A depends on module B. Module A has a create_before_destroy lifecycle policy on it. Module B contains a null resource. The initial creation of the resource using module A goes through fine but any further update runs into cycle error. 

This was tested in terraform 0.12.26 and the dependency was specified using `module_depends_on` variable and there were no issues.

**Terraform files**

Folder Structure
```
- moduleA /
     moduleA.tf
- moduleB /
     moduleB.tf
     test.py
- example /
     example.tf
```
moduleA.tf

```
module test_package {
    source = "../moduleB"
}

resource "aws_sns_topic" "user_updates" {
  lifecycle {
      create_before_destroy = true
  }
  depends_on = [module.test_package]
  name = "user-updates-topic"
}
```
moduleB.tf

```
resource "null_resource" "test_service" {
  triggers = {
        always_run = "${timestamp()}"
    }
  provisioner "local-exec" {
      working_dir = path.module
      interpreter = ["python3"]
      command = "test.py"
  }
}
```

test.py
```
def main():
    print("hello world")
    
if __name__ == "__main__":
    main()
```

example.tf

```
module "test_module" {
  source      = "../moduleA"
}
```

**Steps to Reproduce**
Apply example.tf and it should go through fine. Change the sns topic name in moduleA and run the terraform apply again and it will run into cycle error. 

**Expected Behaviour**
Updates to go through without errors

**Actual Behaviour**
`Error: Cycle: module.test_module.aws_sns_topic.user_updates, module.test_module.aws_sns_topic.user_updates (destroy deposed d55db4bf), module.test_module.module.test_package.null_resource.test_service (destroy deposed 84aabca6), module.test_module.module.test_package (close)`

### PR DESCRIPTION
**Pull request by jbardin**
One of the tenants of the graph transformations is that resource destroy
nodes can only be ordered relative to other resources, and can't be
referenced directly. This was broken by the module close node which
naively connected to all nodes within the module, creating cycles in some
cases when edges are reversed from CreateBeforeDestroy.

Fixes #26166

### CODE REVIEW

File: terraform/transform_module_expansion.go [hunk 1]
```diff
@@ -84,17 +89,17 @@ func (t *ModuleExpansionTransformer) transform(g *Graph, c *configs.Config, pare
 		Config:     c.Module,
 		ModuleCall: modCall,
 	}
-	var v dag.Vertex = n
+	var expander dag.Vertex = n
```
Comment (mildwonkey) [inline terraform/transform_module_expansion.go:92]: I appreciate this clearer variable name!

**Review by mildwonkey — APPROVED** [review]
We'll have to do some digging, I think there are a number of issues open at this point that this can close!

**Comment by Nuru** [comment]
@jbardin wrote:
> One of the tenants of the graph transformations...

I'm confused. Did you mean tenets?

**Comment by jbardin** [comment]
@Nuru, sorry, that was a typo. Thanks

**Pull request merged**